### PR TITLE
add support for imds version 2

### PR DESCRIPTION
**Pull request by junkb**
adds support for imds version 2

### CODE REVIEW

**Comment by tomalok** [comment]
This can be done with `nc` to avoid adding `curl` as an additional dependency...  Here's a test I just did to see if this is doable...
```
$ echo -ne \
  "PUT /latest/api/token HTTP/1.0\r\nX-aws-ec2-metadata-token-ttl-seconds: 60\r\n\r\n" \
  | nc 169.254.169.254 80; echo
HTTP/1.0 200 OK
X-Aws-Ec2-Metadata-Token-Ttl-Seconds: 60
Content-Length: 56
Date: Sat, 18 Apr 2020 19:25:41 GMT
Server: EC2ws
Content-Type: text/plain

AQAEAAgnVBR9zyTEl9Dm1SeoF6fQVwf9WDZuKt2f2YlPA_3LaO08sg==
```

File: tiny-ec2-bootstrap [hunk 1]
```diff
@@ -70,6 +75,7 @@ start() {
 
     [ -d "/var/lib/cloud" ] || mkdir -p /var/lib/cloud
 
+    ebegin "Retrieving metadata service api token"; _get_metadata_token; eend $?
```
Comment (tomalok) [inline tiny-ec2-bootstrap:78]: see `_get_metadata()` comment, above

**Comment by tomalok** [comment]
@junkb -- included in PR #8

**Comment by mcrute** [comment]
Thanks for your contribution @junkb! I'm going to pull this in as #8 with @tomalok's changes and get a release cut to upstream.

**Comment by mcrute** [comment]
The Alpine package has been updated and is [pending merge](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/7561). It should hopefully be in edge within the next day or so and then officially in the next Alpine release.

**Pull request closed without merging**